From 87ba2db064b14bc0f7c7289decd2fea27c644288 Mon Sep 17 00:00:00 2001
From: He Junyan <junyan.he@intel.com>
Date: Tue, 19 Jul 2022 16:27:01 +0800
Subject: [PATCH 205/232] codecparsers: av1: delete the useless
 increment_tile_rows_log2 in tile info.

increment_tile_rows_log2 is only used as the bool flag for parsing tiling log2
number for each while loop, we should not store it in GstAV1FrameHeaderOBU.
---
 gst-libs/gst/codecparsers/gstav1parser.c | 4 ++--
 gst-libs/gst/codecparsers/gstav1parser.h | 2 --
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/gst-libs/gst/codecparsers/gstav1parser.c b/gst-libs/gst/codecparsers/gstav1parser.c
index 5f696fa04..a54010d0e 100644
--- a/gst-libs/gst/codecparsers/gstav1parser.c
+++ b/gst-libs/gst/codecparsers/gstav1parser.c
@@ -2277,11 +2277,11 @@ gst_av1_parse_tile_info (GstAV1Parser * parser, GstBitReader * br,
     min_log2_tile_rows = MAX (min_log2_tiles - parser->state.tile_cols_log2, 0);
     parser->state.tile_rows_log2 = min_log2_tile_rows;
     while (parser->state.tile_rows_log2 < max_log2_tile_rows) {
-      tile_info->increment_tile_rows_log2 = AV1_READ_BIT_CHECKED (br, &retval);
+      gint increment_tile_rows_log2 = AV1_READ_BIT_CHECKED (br, &retval);
       if (retval != GST_AV1_PARSER_OK)
         goto error;
 
-      if (tile_info->increment_tile_rows_log2 == 1)
+      if (increment_tile_rows_log2 == 1)
         parser->state.tile_rows_log2++;
       else
         break;
diff --git a/gst-libs/gst/codecparsers/gstav1parser.h b/gst-libs/gst/codecparsers/gstav1parser.h
index a5f1c761f..2c7106dd3 100644
--- a/gst-libs/gst/codecparsers/gstav1parser.h
+++ b/gst-libs/gst/codecparsers/gstav1parser.h
@@ -1188,7 +1188,6 @@ struct _GstAV1SegmenationParams {
  *   frame. (In other words, all tiles are the same size except for the ones at the right and
  *   bottom edge which can be smaller.) @uniform_tile_spacing_flag equal to 0 means that the
  *   tile sizes are coded.
- * @increment_tile_rows_log2: is used to compute @tile_rows_log2.
  * @width_in_sbs_minus_1: specifies the width of a tile minus 1 in units of superblocks.
  * @height_in_sbs_minus_1: specifies the height of a tile minus 1 in units of superblocks.
  * @tile_size_bytes_minus_1: is used to compute @tile_size_bytes
@@ -1207,7 +1206,6 @@ struct _GstAV1SegmenationParams {
  */
 struct _GstAV1TileInfo {
   guint8 uniform_tile_spacing_flag;
-  gint increment_tile_rows_log2;
   gint width_in_sbs_minus_1[GST_AV1_MAX_TILE_COLS];
   gint height_in_sbs_minus_1[GST_AV1_MAX_TILE_ROWS];
   gint tile_size_bytes_minus_1;
-- 
2.40.1

